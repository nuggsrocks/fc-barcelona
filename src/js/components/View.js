import React from 'react'
import PropTypes from 'prop-types'

export const View = ({ title, text }) => {
  return (
    <section>
      <h1>{title}</h1>
      {
        text.split('\n').filter(x => x !== '').map((paragraph, index) => {
          return <p key={index}>{paragraph}</p>
        })
      }
    </section>
  )
}

View.defaultProps = {
  title: '',
  text: ''
}

View.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string
}
