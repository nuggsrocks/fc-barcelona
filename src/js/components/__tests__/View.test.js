/* global React, render */ 
import { render } from '@testing-library/react'
import { View } from '../View'

describe('<View/>', () => {
    it('should render given text to dom within section element with seperate paragraph elements for each new line', () => {
        const mockText = 'foo bar\nlorem ipsum\n\nbarfoo'

        render(<View text={mockText}/>)

        expect(document.body).toMatchSnapshot()
    })
    it('should render given title to dom within h2 element', () => {
        const mockTitle = 'this is a title'

        render(<View title={mockTitle}/>)

        expect(document.body).toMatchSnapshot()
    })
})