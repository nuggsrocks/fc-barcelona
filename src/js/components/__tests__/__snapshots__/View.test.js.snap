// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View/> should render given text to dom within section element with seperate paragraph elements for each new line 1`] = `
<body>
  <div>
    <section>
      <h1 />
      <p>
        foo bar
      </p>
      <p>
        lorem ipsum
      </p>
      <p>
        barfoo
      </p>
    </section>
  </div>
</body>
`;

exports[`<View/> should render given title to dom within h2 element 1`] = `
<body>
  <div>
    <section>
      <h1>
        this is a title
      </h1>
    </section>
  </div>
</body>
`;
